bin/add_frontend_server.rb

```diff
@@ -98,3 +154,43 @@
 private_dns_name = ec2client.describe_instances({instance_ids: [instance_id],}).reservations[0].instances[0].private_dns_name
 
 puts "Private DNS name #{private_dns_name}"
+
+puts 'Writing new configuration file'
+
+Net::SSH.start('gateway.code.org', username, :password => password) do |ssh|
+  ssh.exec!("knife environment show #{options['environment']} -F json > /tmp/old_knife_config")
+end
+
+Net::SCP.download!('gateway.code.org', username, '/tmp/old_knife_config', '/tmp/knife_config', :ssh => { :password => password })
+
+configuration_json = JSON.parse(File.read('/tmp/knife_config'))
+configuration_json['override_attributes']['cdo-secrets']['app_servers'] = {} if configuration_json['override_attributes']['cdo-secrets']['app_servers'].nil?
```

Comment:
You can write this more concisely as

configuration_json['override_attributes']['cdo-secrets']['app_servers'] ||= {}